### PR TITLE
Allow space keyword in autotype sequence

### LINKED ISSUE #256: Autotype - allow space key
On certain login pages, there's a need to toggle checkbox - usually with the space bar

## Expected Behavior
{SPACE} in autotype sequence should send space key press.

## Current Behavior
On macOS, there's no way to to this.
On Linux (Fedora 25), typing a literal space in the autotype sequence seems to work.

## Possible Solution
Make {SPACE} send the space key.

## Steps to Reproduce (for bugs)
Use {TAB}{TAB}{SPACE}, for example, on any page with a checkbox.

## Your Environment
* KeePassXC 2.1.0 official binary.
* Qt version: 5.7.0
* Operating System and version: macOS Sierra 10.12.3

### PR DESCRIPTION
**Pull request by TheZ3ro**
## Description

Fixes #256 

## How Has This Been Tested?

Manually with autotype.

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ My change requires a change to the documentation and I have updated it accordingly.

### CODE REVIEW

File: src/autotype/AutoType.cpp [hunk 1]
```diff
@@ -396,6 +396,9 @@ QList<AutoTypeAction*> AutoType::createActionFromTemplate(const QString& tmpl, c
     else if (tmplName.compare("enter",Qt::CaseInsensitive)==0) {
         list.append(new AutoTypeKey(Qt::Key_Enter));
     }
+    else if (tmplName.compare("space",Qt::CaseInsensitive)==0) {
+        list.append(new AutoTypeChar(' '));
```
Comment (louib) [inline src/autotype/AutoType.cpp]: @TheZ3ro why not use `Qt::Key_Space`?

Comment (TheZ3ro) [inline src/autotype/AutoType.cpp]: I've tested the `Qt::Key_Space` but returned the error `No such key: keysym=0x0` (on GNU/Linux)
So I opted for `Char(' ')` that works fine.

**Comment by TheZ3ro** [comment]
Ok @louib now should work with  `Qt::Key_Space` on all platforms. Someone can test on windows?

**Comment by phoerious** [comment]
I tested it on Windows and it works.

**Pull request merged**